### DISCUSSION TITLE
20240118 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#78] [Meetings]
Time: 8:30 GMT+0  Canary Islands Time (9:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@WarmBeer) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by cgbosse** [comment]
## Torrust Project Organization:

- Project structuring
- New views for different use 
a. [Website](https://github.com/orgs/torrust/projects/10/views/10)
b. [Roadmap WIP](https://github.com/orgs/torrust/projects/10/views/6)



- Discussion and collaboration with @josecelano to establish what to include in the release v3.0.0
- Roadmap WIP
- Labels


Note: "Good first issue" -- changed description and deleted the "First time contribution" label.
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/encouraging-helpful-contributions-to-your-project-with-labels

## Good practices to become standard:

Weekly meeting notes:
---------------------------
- To be done before each meeting.
- List the different tasks you did working on the projects.
- List if you had any challenges where you need input or help.
- List the ideas that you would like to have a discussion about, why, what and how these may be relevant or important.
- Indicate if you spent any time learning anything with potential relevance for the other members of the team.
- What other things of value you did, why, what and how these may be relevant or important.

Why: 
The purpose is that you condense the above information in such a way that it serves you as an agenda to make your presentation. list but also gives a good overview of how you worked on projects, spent time learning, etc. to anyone who cannot attend the meeting. If you encountered any challenges that you need help with. 

Creating complex Issues with subtasks:  
-----------------------------------------------
1. Label as EPIC
2. Label other relevant label
3. Indicate the subtasks as check boxes so that this reflects in the overall progress of the issue.
 
Example: https://github.com/torrust/torrust-tracker/issues/160


 Toggling: 
------------------------------------------------
1. Assign always the correct project do not leave it blank, there are projects created for this in Toggl.
2. In the description:
- indicate the repository in square brackets such as for example: [Index], [GUI], [Website], [Tracker], [Boken], etc.
- indicate the issue, PR, or discussion  number such as [#999], [#998], etc.
- add the description of the discussion or
- indicate the action separated by a hypen make it clear and simple such as for example updated, reviewed, coding, commented, etc. 

Examples:
![image](https://github.com/nautilus-cyberneering/.github/assets/15330600/13efec7e-6860-4316-9f5e-f3018f8e4888)


## General discussion:

- Priorizing: https://github.com/orgs/nautilus-cyberneering/discussions/66#discussioncomment-7534160

**Comment by mickvandijke** [comment]
## Torrust Tracker

I have been working on adding a configurable memory limit to the torrent repository: https://github.com/torrust/torrust-tracker/issues/567

**Comment by josecelano** [comment]
### Torrust Roadmap

I've been adding issues to milestones together with @cgbosse.

https://github.com/orgs/torrust/projects/10/views/1

### Torrust Website

- PR review: [#[129](https://github.com/torrust/torrust-website/pull/129)] [Add share buttons at the top of each blog post](https://github.com/torrust/torrust-website/pull/129)

### Torrust Tracker

PR reviews:

- PR review: [#[623](https://github.com/torrust/torrust-tracker/pull/623)] [Extract Config from Tracker and Heath Check](https://github.com/torrust/torrust-tracker/pull/623)

Issues:

- Issue: [#[604](https://github.com/torrust/torrust-tracker/issues/604)] [Add timeout to the http_health_check.rs binary](https://github.com/torrust/torrust-tracker/issues/604) 

- Issue: [#[614](https://github.com/torrust/torrust-tracker/issues/614)] [Add a new HTTP tracker client binar](https://github.com/torrust/torrust-tracker/issues/614)

- Issue: [#[609](https://github.com/torrust/torrust-tracker/issues/609)] [Add timeout to UDP tracker request processing](https://github.com/torrust/torrust-tracker/issues/609)

- Issue: [#[625](https://github.com/torrust/torrust-tracker/issues/625)] [Add a new UDP tracker client binary ](https://github.com/torrust/torrust-tracker/issues/625). **WIP**. I'm working on this issue but I'm having a strange problem. It seems the client does not actually send the request to the server, but it returns the same package, like an echo server.

- Issue: [#[591](https://github.com/torrust/torrust-tracker/issues/591)] [Error starting UDP tracker: Failed to install stop signal: channel closed](https://github.com/torrust/torrust-tracker/issues/591). 

Issue #591:

I fixed that problem but we have it again:

- [ ] https://github.com/torrust/torrust-tracker/issues/626

After merging:

- https://github.com/torrust/torrust-tracker/pull/623

@da2ce7 should we revert the PR?

I'm planning to add [E2E tests](https://github.com/torrust/torrust-tracker/issues/620) to avoid regressions, especially for the app bootstrapping because we do not have any tests.

**Comment by grmbyrn** [comment]
## Torrust Website
* https://github.com/torrust/torrust-website/pull/129
* New blog post: An Introduction to Torrust: Enhancing the BitTorrent Protocol Ecosystem
* Link in Hero to live demo
* Research other blogs. Eg: [a](https://www.creativebloq.com/) [b](https://thedsgnblog.com/) [c](https://www.webdesignerdepot.com/) [d](https://css-tricks.com/) [e](https://design-milk.com/) [f](https://sidebar.io/) [g](https://www.smashingmagazine.com/)
* Different ideas for how the page should look: https://miro.com/app/board/uXjVN7YuMmY=/
* Fix spacing on table of contents - blog post on larger screens
* Fix bug on strap at top of page

## Research/Revision
* Docker

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/421

### Training/Research

- CSS (W3shools tutorials, Flexbox model, transitions)
